### PR TITLE
[tests-only][full-ci] add test to check acivities by sharee

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds test where a sharee checks api activities.

``` feature
Scenario: sharee tries to check the activities of a shared folder using share mount-point id
Scenario: sharee tries to check the activities of unshared file
Scenario: sharee tries to check the activities of a shared folder using folder id
```

## Related Issue

- Related to : https://github.com/owncloud/ocis/issues/9543

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by PrajwolAmatya** [comment]
Blocked until this bug https://github.com/owncloud/ocis/issues/9849 is fixed.

**Comment by prashant-gurung899** [comment]
> Blocked until this bug #9849 is fixed.

Unblocking this PR as this PR adds test for checking activities of a shared folder using mountpoint share id.
For bug :https://github.com/owncloud/ocis/issues/9849 i.e. i will cover it in new PR after it gets fixed.

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 1]
```diff
@@ -1076,3 +1076,48 @@ Feature: check share activity
         }
       }
       """
+
+  @issue-9849
+  Scenario: sharee tries to check activities of folder using share mountpoint id
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1081]: issue is not valid for this scenario
```suggestion

  Scenario: sharee tries to check activities of folder using share mountpoint id
```

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 2]
```diff
@@ -1076,3 +1076,48 @@ Feature: check share activity
...
+            "uniqueItems": true,
+            "items": {
+            }
+          }
+        }
+      }
+      """
+
+  @issue-9849
+  Scenario: sharee tries to check activities of another unshared file
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1114]: this issue might be the one for this scenario
```suggestion
  @issue-9860
  Scenario: sharee tries to check activities of another unshared file
```

File: tests/acceptance/features/apiActivities/shareActivities.feature (hunk 1 above)
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1081]:
```suggestion
  Scenario: sharee tries to check the activities of a shared folder using share mount-point id
```

File: tests/acceptance/features/apiActivities/shareActivities.feature (hunk 2 above)
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1114]:
```suggestion
  Scenario: sharee tries to check the activities of unshared file
```

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 3]
```diff
@@ -1076,3 +1076,48 @@ Feature: check share activity
...
+    Given user "Alice" has uploaded file with content "another ownCloud test text file" to "anotherTextfile.txt"
+    And user "Alice" has sent the following resource share invitation:
+      | resource        | textfile.txt |
+      | space           | Personal     |
+      | sharee          | Brian        |
+      | shareType       | user         |
+      | permissionsRole | Viewer       |
+    When user "Brian" tries to list the activities of unshared file "anotherTextfile.txt" of user "Alice" from space "Personal" using the Graph API
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1122]:
```suggestion
    When user "Brian" tries to list the activities of file "anotherTextfile.txt" owned by user "Alice" using the Graph API
```

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 4]
```diff
@@ -1076,3 +1076,48 @@ Feature: check share activity
...
+      | resource        | /FOLDER  |
+      | space           | Personal |
+      | sharee          | Brian    |
+      | shareType       | user     |
+      | permissionsRole | Editor   |
+    And user "Brian" has a share "/FOLDER" synced
+    And user "Brian" has uploaded file with content "some data" to "Shares/FOLDER/newfile.txt"
+    And user "Brian" has uploaded file with content "edited data" to "Shares/FOLDER/newfile.txt"
+    And user "Brian" has deleted file "Shares/FOLDER/newfile.txt"
+    When user "Brian" lists the activities for folder "FOLDER" of space "Shares" using the Graph API
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1093]: could you update this step?
```suggestion
    When user "Brian" lists the activities of folder "FOLDER" from space "Shares" using the Graph API
```

File: tests/acceptance/bootstrap/GraphContext.php [hunk 5]
```diff
@@ -2808,7 +2808,39 @@ public function getPermissionRoleDefinitionUsingGraphAPI(string $user, string $p
...
+	 * @When user :user tries to list the activities of unshared file :file of user :user1 from space :spaceName using the Graph API
+	 *
+	 * @param string $user
+	 * @param string $file
+	 * @param string $user1
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 */
+	public function userTriesToListActivitiesOfUnsharedFile(string $user, string $file, string $user1, string $spaceName): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2842]:
```suggestion
	 * @param string $owner
	 * @param string $spaceName
	 *
	 * @return void
	 */
	public function userTriesToListActivitiesOfUnsharedFile(string $user, string $file, string $owner, string $spaceName): void {
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiActivities/shareActivities.feature (hunk 3 above)
Comment (prashant-gurung899) [reply tests/acceptance/features/apiActivities/shareActivities.feature:1122]: @saw-jan I think we need the spaceName as well to get the resourceId in this step. What do you think?

File: tests/acceptance/features/apiActivities/shareActivities.feature (hunk 4 above)
Comment (prashant-gurung899) [reply tests/acceptance/features/apiActivities/shareActivities.feature]: @saw-jan 
May be we should update this step as well?
``` feature
When user "Alice" lists the activities for folder "New Folder" of space "Personal" with depth "-1" using the Graph API
```

Comment (saw-jan) [reply tests/acceptance/features/apiActivities/shareActivities.feature]: yes, please

File: tests/acceptance/features/apiActivities/shareActivities.feature (hunk 3 above)
Comment (saw-jan) [reply tests/acceptance/features/apiActivities/shareActivities.feature]: ahh, then we can accommodate space name as well
`When user "Brian" tries to list the activities of file "anotherTextfile.txt" from space "Personal" owned by user "Alice" using the Graph API`

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 6]
```diff
@@ -1076,3 +1076,47 @@ Feature: check share activity
...
+  @issue-9860
+  Scenario: sharee tries to check the activities of unshared file
+    Given user "Alice" has uploaded file with content "another ownCloud test text file" to "anotherTextfile.txt"
+    And user "Alice" has sent the following resource share invitation:
+      | resource        | textfile.txt |
+      | space           | Personal     |
+      | sharee          | Brian        |
+      | shareType       | user         |
+      | permissionsRole | Viewer       |
+    When user "Brian" tries to list the activities of file "anotherTextfile.txt" owned by user "Alice" from space "Personal" using the Graph API
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1105]:
```suggestion
    When user "Brian" tries to list the activities of file "anotherTextfile.txt" from space "Personal" owned by user "Alice" using the Graph API
```

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 7]
```diff
@@ -1076,3 +1076,47 @@ Feature: check share activity
...
+    Then the HTTP status code should be "403"
+
+  @issue-9849
+  Scenario: sharee tries to check the activities of a shared folder using folder id
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1109]:
```suggestion
  Scenario: sharee tries to check the activities of a shared folder using file-id
```

File: tests/acceptance/features/apiActivities/shareActivities.feature [hunk 8]
```diff
@@ -1076,3 +1076,47 @@ Feature: check share activity
...
+    And user "ALice" has uploaded file with content "edited data" to "FOLDER/newfile.txt"
+    And user "ALice" has deleted file "FOLDER/newfile.txt"
+    And user "Alice" has sent the following resource share invitation:
+      | resource        | /FOLDER  |
+      | space           | Personal |
+      | sharee          | Brian    |
+      | shareType       | user     |
+      | permissionsRole | Editor   |
+    And user "Brian" has a share "/FOLDER" synced
+    When user "Brian" lists the activities of folder "FOLDER" from space "Shares" using the Graph API
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/shareActivities.feature:1121]: These scenarios must not use the same When step:
```feature
Scenario: sharee tries to check the activities of a shared folder using share mount-point id
Scenario: sharee tries to check the activities of a shared folder using folder id
```

Comment (saw-jan) [reply tests/acceptance/features/apiActivities/shareActivities.feature:1121]: also put them together.

Comment (prashant-gurung899) [reply tests/acceptance/features/apiActivities/shareActivities.feature:1121]: So, should we make a new step to list activities of shared file/folder using share mount-point id?

File: tests/acceptance/bootstrap/GraphContext.php [hunk 9]
```diff
@@ -2808,7 +2808,39 @@ public function getPermissionRoleDefinitionUsingGraphAPI(string $user, string $p
...
+			$this->featureContext->getStepLineRef(),
+			$user,
+			$this->featureContext->getPasswordForUser($user),
+			$resourceId
+		);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @When user :user tries to list the activities of file :file owned by user :user1 from space :spaceName using the Graph API
```
Comment (PrajwolAmatya) [inline tests/acceptance/bootstrap/GraphContext.php:2833]:
```suggestion
	 * @When user :user tries to list the activities of file :file owned by user :owner from space :spaceName using the Graph API
```

File: tests/acceptance/features/apiActivities/shareActivities.feature (hunk 8 above)
Comment (saw-jan) [reply tests/acceptance/features/apiActivities/shareActivities.feature:1121]: yeah

File: tests/acceptance/bootstrap/GraphContext.php [hunk 10]
```diff
@@ -2819,6 +2819,55 @@ public function userListsTheActivitiesForResourceOfSpaceUsingTheGraphAPI(string
 		$this->featureContext->setResponse($response);
 	}
 
+	/**
+	 * @When user :user tries to list the activities of folder :folder from space :spaceName using share mount-point id using the Graph API
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2823]: space should be always `Shares` for this step
```suggestion
	 * @When user :user tries to list the activities of folder :folder with share mount-point-id using the Graph API
```

File: tests/acceptance/bootstrap/GraphContext.php [hunk 11]
```diff
@@ -2819,6 +2819,55 @@ public function userListsTheActivitiesForResourceOfSpaceUsingTheGraphAPI(string
...
+	 *
+	 * @param string $user
+	 * @param string $folder
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 */
+	public function userListsTheActivitiesOfFolderFromSpaceUsingShareMountPointIdUsingTheGraphApi(string $user, string $folder, string $spaceName): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2831]:
```suggestion
	public function userListsTheActivitiesOfFolderFromSpaceWithShareMountIdPointIdUsingTheGraphApi(string $user, string $folder, string $spaceName): void {
```

File: tests/acceptance/bootstrap/GraphContext.php [hunk 12]
```diff
@@ -2819,6 +2819,54 @@ public function userListsTheActivitiesForResourceOfSpaceUsingTheGraphAPI(string
...
+	 * @When user :user tries to list the activities of file :file from space :spaceName owned by user :owner using the Graph API
+	 *
+	 * @param string $user
+	 * @param string $file
+	 * @param string $owner
+	 * @param string $spaceName
+	 *
+	 * @return void
+	 */
+	public function userTriesToListActivitiesOfUnsharedFile(string $user, string $file, string $owner, string $spaceName): void {
```
Comment (amrita-shrestha) [inline tests/acceptance/bootstrap/GraphContext.php:2858]:
```suggestion
	public function userTriesToListActivitiesOfFileFromSpaceOwnedByUserUsingTheGraphApi(string $user, string $file, string $owner, string $spaceName): void {
```

File: tests/acceptance/bootstrap/GraphContext.php [hunk 13]
```diff
@@ -2823,6 +2823,54 @@ public function userListsTheActivitiesForResourceOfSpaceUsingTheGraphAPI(string
...
 
+	/**
+	 * @When user :user tries to list the activities of folder :folder with share mount-point id using the Graph API
+	 *
+	 * @param string $user
+	 * @param string $folder
+	 *
+	 * @return void
+	 */
+	public function userListsTheActivitiesOfFolderFromSpaceWithShareMountIdPointIdUsingTheGraphApi(string $user, string $folder): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2834]: let's name the step defs as close as possible to step name
```suggestion
	public function userTriesToListTheActivitiesOfFolderWithShareMountIdPointIdUsingTheGraphApi(string $user, string $folder): void {
```

Comment (prashant-gurung899) [reply tests/acceptance/bootstrap/GraphContext.php:2834]: Sure.

**Review by saw-jan — APPROVED** [review]
:+1: :sparkles:

**Pull request merged**